src/dotnet/Program.cs

```diff
@@ -138,6 +141,16 @@ internal static int ProcessArgs(string[] args, ITelemetry telemetryClient)
 
         }
 
+        /// <summary>
+        /// Performs initialization that must be done before anything else in the process.
```

Comment:
nit: I don't think this comment is necessary, I would infer this from the method name.